tests/sqlparser_redshift.rs

```diff
@@ -353,3 +380,23 @@ fn test_parse_json_path_from() {
         _ => panic!(),
     }
 }
+
+#[test]
+fn test_parse_select_numbered_columns() {
+    // An alias starting with a number
+    redshift_and_generic().verified_stmt(r#"SELECT 1 AS "1" FROM a"#);
+    redshift_and_generic().verified_stmt(r#"SELECT 1 AS "1abc" FROM a"#);
+}
+
+#[test]
+fn test_parse_nested_quoted_identifier() {
+    redshift().verified_stmt(r#"SELECT 1 AS ["1"] FROM a"#);
+    redshift().verified_stmt(r#"SELECT 1 AS [ " 1 " ]"#);
```

Comment:
Ah I think the expected behavior would be that the parser strips out whitespace:
```rust
    redshift().one_statement_parses_to(
        r#"SELECT 1 AS [ " 1 " ]"#,
        r#"SELECT 1 AS [" 1 "]"#,
    );
```

Reply:
It depends on what we want to achieve. I have implemented parsing of nested quotations, guided by the test case [brackets_over_db_schema_table_name_with_whites_paces](https://github.com/apache/datafusion-sqlparser-rs/blob/main/tests/sqlparser_redshift.rs#L69). For example, the following query:
```
SELECT [   col1  ] FROM [  test_schema].[ test_table]
```
is successfully parsed using `verified_stmt`.

I don't have a strong opinion on this implementation and can adjust the parsing if needed.

Reply:
Oh yeah I think that the linked test would be incorrect as well? from what I could tell and tested the following seemed to all be equivalent in redshift
```sql
SELECT 1 AS foo;
SELECT 1 AS [ foo ];
SELECT 1 AS [ "foo" ];
```
so that whitespace within the brackets isn't part of the identifier. Then it would be incorrect to have an identifier in the AST that contains the string ` foo ` for any of the above.

On the other hand this would actually be referencing an identifier containing whitespace
```sql
SELECT 1 AS [" foo "];
```

Reply:
I have updated the PR to trim spaces in cases of nested quotations, e.g., `[  "foo"  ]` -> `["foo"]`. Implementing similar behavior for cases like `[  foo  ]` would require more extensive changes to the existing code. In my opinion, this could introduce some risk, at least in the context of the current PR.